### DISCUSSION TITLE
How to view restic log?

### DISCUSSION THREAD

**Discussion by jontyms** [cupcakearmy/autorestic#134] [Q&A] [answered]
How to view the restic log?
Thanks

**Comment by cupcakearmy** [comment]
What do you mean? the cli output?

**Reply by jontyms** [reply]
Yes I would like to see the output of the restic command

**Reply by cupcakearmy** [reply]
Have you tried to add the verbose flag? ` -v, --verbose             verbose mode`

**Reply by jontyms** [reply]
Yes when I use -v it says
`Executing: /usr/bin/restic backup --exclude-if-present .exclude_from_backup --tag piconf --tag ar:location:con
fig21 /home/ubuntu/config `

but it doesn't output the restic output.
in this case restic is giving a error (exit status 3) and I can't see what it is

**Reply by cupcakearmy** [reply]
Interesting, because error and logs from std are piped straigh to the console.
So restic might exit immerdiately? Could you try running restic alone?
I just realized it would be cool to have a `autorestic env` command to ouput all the envs used

**Reply by jontyms** [reply]
when I run restic on its own it outputs `Warning: failed to read all source data during backup` but also saves a snapshot.
autorestic doesn't output anything and reports failure [![asciicast](https://asciinema.org/a/b51JZKLhpaxgtpXqr9j4VOHmM.svg)](https://asciinema.org/a/b51JZKLhpaxgtpXqr9j4VOHmM)
restic snapshot shows that a snapshot is created

**Reply by cupcakearmy** [reply]
Still not sure, could you maybe post your autorestic config? (WIth redacted keys of course)

**Reply by jontyms** [reply]
```
---
version: 2
locations:
  config21:
    from:
      - /home/ubuntu/config
    to:
      - remoteb2
    hooks:
      before:
        - curl https://hc-ping.com//pi-config-backup/start
      failure:
        - curl https://hc-ping.com//pi-config-backup/fail
      success:
        - curl https://hc-ping.com//pi-config-backup
          #    cron: 0 * * * *
    options:
      backup:
        exclude-if-present:
        - .exclude_from_backup
        tag:
        - piconf
      forget:
        keep-daily:
           - 7
        keep-monthly:
           - 2
        keep-weekly:
           - 4
        keep-yearly:
           - 1
backends:
  remoteb2:
    type: b2
    path: jontybackup
    key: 
    env:
      b2_account_id: 
      b2_account_key:

**Reply by cupcakearmy** [reply]
I just saw there is a bug that error logs sometimes can't be displayed. It will be fixed in #136

**Comment by jontyms** [comment, marked as answer]
```
config21@remoteb2:
error: open /home/ubuntu/config/notification.txt: permission denied
Warning: at least one source file could not be read
exit status 3
```

so when restic can't access some files it gives exit code 3, autorestic runs failure hook
But restic still creates a snapshot
